packages/flutter_tools/lib/src/web/compile.dart

```diff
@@ -36,10 +37,16 @@ Future<void> buildWeb(
   final bool hasWebPlugins = (await findPlugins(flutterProject))
     .any((Plugin p) => p.platforms.containsKey(WebPlugin.kConfigKey));
   final Directory outputDirectory = globals.fs.directory(getWebBuildDirectory());
-  if (outputDirectory.existsSync()) {
-    outputDirectory.deleteSync(recursive: true);
+  try {
+    ErrorHandlingFileSystem.deleteIfExists(outputDirectory, recursive: true);
     outputDirectory.createSync(recursive: true);
+  } on FileSystemException catch (err) {
+    globals.logger.printError(
```

Comment:
I think I'd rather crash than keep on trucking. The latter could lead to even harder to debug situations.

Since this function is async, can we print the issue and wait until the directory is unlocked? Pseudo-code:

```dart
bool deleted = false;
bool warningPrinted = false;

while (!deleted) {
  try {
    delete();
    deleted = true;
  } on FileSystemException {
    if (!warningPrinted) {
      globals.logger.printError('Directory $outputDirectory is locked by another program. Waiting for the directory to be unlocked.');
      warningPrinted = true;
      await Future.delayed(Duration(milliseconds: 50));
    }
  }
}

if (warningPrinted) {
  globals.logger.printInfo('Directory $outputDirectory was unlocked and deleted. Continuing.');
}
```

Reply:
This could definitely work, I was leaning towards the "quick fix" angle :)

Reply:
though I think the problem with this is that we will probably delete most of the files successfully, so that leaves a partially invalid state

Reply:
actually ... lets do a partial revert for now. this is causing crashes and I don't quite have the time I need for a long term solution